src/controllers/bmdashboard/bmConsumableController.js

```diff
@@ -1,45 +1,101 @@
 const bmConsumableController = function (BuildingConsumable) {
-    const fetchBMConsumables = async (req, res) => {
-      try {
-        BuildingConsumable
-          .find()
-          .populate([
-            {
-              path: 'project',
-              select: '_id name',
+  const fetchBMConsumables = async (req, res) => {
+    try {
+      BuildingConsumable
```

Comment:
Code looks good, just a question. Is there a reason why we use promise chaining instead of async await? We have the async function but then use promise chaining. Is that just the codebase conventions?

Reply:
@Abi-Liu no particular reason, there is no convention for how async functions should be handled in this org. You could write **await BuildingConsumable.find().then(callback).catch(error)** inside an async function just as well. The MongoDB documentation mostly has the try/catch syntax, I think that's why most of HGNRest code is using that too. I was referencing some very similar existing code that works well so I didn't change it.

Reply:
Gotcha, thanks for clearing that up